test1Tile.py

```diff
@@ -0,0 +1,212 @@
+                    if curBand.find('i') >= 0: objind['ip'] = ext     
+                    if curBand.find('z') >= 0: objind['zp'] = ext     
+                    self.objext[curObj] = objind
+
+
+            ext+=1        
+        fits.close()
+        " loop on objects and create PNG images "
+        fits = fitsio.FITS(fitsfile,'rw')
+        interesting=[ "3017263885", "3017249430", "3017265746", "3017263357", "3017255895", "3017261463", "3017263527"]
+
+        for objectN in self.objects:
+            objext = self.objext[objectN]
+            outDir = "./gallery/"
+            if not os.path.exists(outDir):
+                os.makedirs(outDir)
+            imdict = {}
+            sigdict= {}
+            psfdict= {}
+
+            for s in ['g','r','i','z']:
+                sw=s+"w"
+                sp=s+"p"
+                a=90
+
+                ext=objext[s]
+                header = fits[ext].read_header()
+                imdict[s] = fits[ext].read()[a:-a,a:-a]
+
+                ext=objext[sw]
+                header = fits[ext].read_header()
+                sigdict[s] = (fits[ext].read()[a:-a,a:-a])**-2
+
+                ext=objext[sp]
+                header = fits[ext].read_header()
+                psfdict[s] = fits[ext].read()
+            print objectN
+            #["3017249430"]
+
+            #BR=BlueRings(imdict,sigdict,psfdict)
+            
+            if objectN in interesting or int(objectN)>3017265667:
+                BR=BlueRings(imdict,sigdict,psfdict)
+                inter=BR.plot()
+                if inter!="":interesting.append(objectN)
+                print "[",
+                for inter in interesting:
+                    print "\"%s\","%inter,
+                print"]"
+
+
+            """
+            plt.subplot(131)
+            color = colorImage.ColorImage()
+            colorimage = color.createModel(imdict['g'],imdict['r'],imdict['i'])
+            plt.imshow(colorimage,interpolation="none")
+            plt.subplot(132)
+
+            #psfmode="crossconvolve"
+            #RF=RingFinder(imdict['g'],sigdict['i']**-2,sigdict['g']**-2,imdict['i'],psfdict['g'],psfdict['i'],0.265,1e12,1e12,visualize=False,psfmode=psfmode)
+            #plt.imshow(RF.Dshow,interpolation="none")
+            #plt.show()
+
+            BR=BlueRings(imdict,sigdict,psfdict)
+
+            colorimage = color.colorize(BR.subdict['g'],BR.subdict['r'],BR.subdict['i'])
+            plt.imshow(colorimage,interpolation="none")
+
+            plt.subplot(133)
+            BR.subdict['g'][BR.subdict['g']<0]=0
+            plt.imshow(BR.subdict['g'],interpolation="none")
+
+
+            plt.draw()
+            raw_input()
+            """
+
+if __name__ == "__main__":
+    dir_input = "/home/ttemp/BlueRings/data/"
+    file_input = "DES0005-0041_cutouts.fits"
+    fitsfile = dir_input+file_input
+
+    tile=RingFindforTile()
+    tile.produce(fitsfile)
+exit() 
+
+
+a=40
+for j in range(100):
```

Comment:
This code looks too useful to be left in `main`. How about we make a `cutout` class that contains all these image, sig and psf arrays, and pass `cutout` objects around instead of arrays? That cutout object could also have a method to return composite images of itself. If you don't feel like doing this now, you could issue the refactoring work instead - but I think we'll get a lot of value out of using `cutout` objects.